### PR TITLE
Add the boilderplate atom render plugin from the emfxatom gem

### PR DESCRIPTION
**Pull request by amzn-rhhong**
1. Added an editorsystemcomponent for emfxatom gem
2. Added a boilderplate code for atom renderplugin
3. Change the singleton pattern in emfx studio manager to use az::interface. This is because we need to share the emfx studio manager between emfx.editor.dll and emfx_atom.editor.dll.
4. Added a RegisterPlugin event for the animation systemNotificationBus so other system could register custom emfx studio plugins.

### CODE REVIEW

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/Editor/EditorSystemComponent.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+#include <Editor/EditorSystemComponent.h>
+#include <EMStudio/AtomRenderPlugin.h>
+
+#include <EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/EMStudioManager.h>
+
+namespace AZ
+{
+    namespace EMotionFXAtom
+    {
```
Comment (amzn-jillich) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/Editor/EditorSystemComponent.cpp:17]: What about using namespace AZ::EMotionFXAtom for new files to reduce width?

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AtomRenderPlugin.h [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            CLASS_ID = 0x32b0c04d
+        };
+        AtomRenderPlugin();
+        ~AtomRenderPlugin();
+
+        // Plugin information
+        const char* GetCompileDate() const override
+        {
+            return MCORE_DATE;
+        }
```
Comment (amzn-jillich) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AtomRenderPlugin.h:34]: Remove? I don't think this is useful and am hoping we can get rid of that at a later stage generally.

Comment (amzn-jillich) [reply Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AtomRenderPlugin.h:34]: Can you also bring the implementations to the .cpp file please?

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/EMStudioManager.cpp [hunk 3]
```diff
@@ -105,6 +100,8 @@ namespace EMStudio
 
         // log some information
         LogInfo();
+
+        AZ::Interface<EMStudioManager>::Register(this);
```
Comment (amzn-jillich) [inline Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/EMStudioManager.cpp:104]: Ah, so this did the trick that we were talking about yesterday.

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/EMStudioManager.cpp [hunk 4]
```diff
@@ -422,6 +421,12 @@ namespace EMStudio
     }
 
 
+    EMStudioManager* EMStudioManager::GetInstance()
+    {
+        return AZ::Interface<EMStudioManager>().Get();
+    }
+
+
```
Comment (amzn-jillich) [inline Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/EMStudioManager.cpp:429]: Nit: One newline too much?

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/EMStudioManager.h [hunk 5]
```diff
@@ -153,31 +157,41 @@ namespace EMStudio
...
+        return EMStudioManager::GetInstance()->GetNotificationWindowManager();
+    }
+    MCORE_INLINE MotionEventPresetManager* GetEventPresetManager()
+    {
+        return EMStudioManager::GetInstance()->GetEventPresetManger();
+    }
+    MCORE_INLINE CommandSystem::CommandManager* GetCommandManager()
+    {
+        return EMStudioManager::GetInstance()->GetCommandManager();
+    }
```
Comment (amzn-jillich) [inline Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/EMStudioManager.h:196]: Please move these over to the .cpp file. We also don't need them inlined.

**Review by amzn-jillich — APPROVED** [review]

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/Editor/EditorSystemComponent.cpp (hunk 1 above)
Comment (amzn-rhhong) [reply Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/Editor/EditorSystemComponent.cpp:17]: good call

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/EMStudioManager.cpp (hunk 3 above)
Comment (amzn-rhhong) [reply Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/EMStudioManager.cpp:104]: Yes, this is the way to "transfer" static object between dlls/

**Pull request merged**